cygnus-twitter/src/main/java/com/telefonica/iot/cygnus/sources/TwitterSource.java

```diff
@@ -0,0 +1,316 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.telefonica.iot.cygnus.sources;
+
+import org.apache.flume.Context;
+import org.apache.flume.Event;
+import org.apache.flume.EventDrivenSource;
+import org.apache.flume.channel.ChannelProcessor;
+import org.apache.flume.conf.Configurable;
+import org.apache.flume.event.EventBuilder;
+import org.apache.flume.source.AbstractSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import twitter4j.*;
+import twitter4j.conf.ConfigurationBuilder;
+
+import java.nio.charset.Charset;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class TwitterSource extends AbstractSource
+        implements EventDrivenSource, Configurable {
+
+    private TwitterStream twitterStream;
+    private List<Event> eventBatch = new ArrayList<Event>();
+
+    private boolean have_filters = false;
```

Comment:
Fixed in 529d171
My IDE (IntelliJ) crashed when applying the xml config file for checkstyle. I've finally run it using a different plugin.